tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/let/LetExpressionTest.java

```diff
@@ -54,9 +54,9 @@ public void testLetExpressionNegative() {
         BAssertUtil.validateError(negativeResult, i++, "redeclared symbol 'x'", 19, 21);
         BAssertUtil.validateError(negativeResult, i++, "undefined symbol 'y'", 23, 27);
         BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'string'", 27, 25);
-        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'string'", 28, 13);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found 'string'", 28, 39);
         BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'string', found 'int'", 29, 28);
-        BAssertUtil.validateError(negativeResult, i, "incompatible types: expected 'string', found 'int'", 30, 16);
+        BAssertUtil.validateError(negativeResult, i, "incompatible types: expected 'string', found 'int'", 30, 42);
```

Comment:
Shall we add some negative tests for this?

Reply:
@dulajdilshan Can you give some examples? For the specific case of #26906, we had a compilation error. Therefore, adding the 
```
public function testAnonymousRecordWithLetExpression() {
    record {
        int i;
        int j?;
    } rec = let int v = 1 in {i: v};

    assertTrue(rec.i == 1, "rec.i == 1");
}
``` 
to the `let-expression-test.bal` file ensures that the compilation no longer fails (otherwise, test initialization fails). Apart from that, what specific cases can we add?

Reply:
Since the assignability is addressed by #27596, negative tests seems to be not required.

Reply:
We can add a negative test like

```ballerina
    record {
        byte i;
        int j?;
    } rec = let int v = 160 in {i: v};
```

Without this fix, the error was something like
```
incompatible types: expected 'record {| byte i; int j?; anydata...; |}', found 'record {| int i; |}'
```

With this fix I believe it'll be something like
```
incompatible types: expected 'byte', found 'int'
```

Reply:
Yes. That's the error we get. Added.